### ISSUE TITLE
Errors raised for List[TypedDict] arguments

### ISSUE THREAD

**Issue by AGiantSquid** [microsoft/pyright#996]
I noticed that functions can pass custom `TypedDict` instances to other functions just fine, but cannot do the same for a `List` with the same type.

Consider the following custom `TypedDict` type:
```python
from typing import TypedDict, List

class DemoType(TypedDict):
    float_field: float
```

Let's say I have a generic function that can do something with TypedDicts:
```python
def dict_func(arg: TypedDict):
    # do work
    pass
```

If I create a function that needs to call this generic function with a custom `TypedDict`, pyright does not complain:

```python

def call_dict_func(arg: DemoType):
    # accept an argument of type DemoType, and feed to dict_func, which expects a TypedDict 
    return dict_func(arg)
```

If I try and do the same thing with a `List` of `TypedDict`s, I get an error:

```python

def list_dict_func(arg: List[TypedDict]):
    # do work
    pass


def call_list_dict_func(arg: List[DemoType]):
     # accept an argument that is a List of DemoType, and feed to dict_func, which expects a List of TypedDict
    return list_dict_func(arg)
```

The error:
```
Argument of type "List" cannot be assigned to parameter "arg" of type "List" in function "list_dict_func"
  TypeVar "_T" is invariant
    "DemoType" is incompatible with "TypedDict"Pyright (reportGeneralTypeIssues)
```

Interestingly, I do not get an error if I create the list in the function call, but I do get an error if I create the list before the function call:

```python
def call_list_dict_func_two(arg: DemoType):

    list_dict_func([arg])  # no error

    listified = [arg]
    list_dict_func(listified)  # this errors
```

Full Snippet:
```python
from typing import TypedDict, List


class DemoType(TypedDict):
    float_field: float


def dict_func(arg: TypedDict):
    # do work
    pass


def call_dict_func(arg: DemoType):
    return dict_func(arg)  # no error


def list_dict_func(arg: List[TypedDict]):
    # do work
    pass


def call_list_dict_func(arg: List[DemoType]):
    return list_dict_func(arg)  # error


def call_list_dict_func_two(arg: DemoType):
    list_dict_func([arg])  # no error
    listified = [arg]
    list_dict_func(listified)  # error
```

Screenshot of errors in VSCode:
![image](https://user-images.githubusercontent.com/26531374/92070269-67a91900-ed60-11ea-8368-7fbc659af7b9.png)

![image](https://user-images.githubusercontent.com/26531374/92070311-7f809d00-ed60-11ea-9eed-bbc338854868.png)

VSCode Info:
```
CPUs | Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz (8 x 1896)
-- | --
GPU Status | 2d_canvas: enabledflash_3d: enabledflash_stage3d: enabledflash_stage3d_baseline: enabledgpu_compositing: enabledmultiple_raster_threads: enabled_onoop_rasterization: disabled_offprotected_video_decode: enabledrasterization: enabledskia_renderer: disabled_off_okvideo_decode: enabledviz_display_compositor: enabled_onviz_hit_test_surface_layer: disabled_off_okwebgl: enabledwebgl2: enabled
Load (avg) | undefined
Memory (System) | 15.82GB (4.11GB free)
Process Argv |  
Screen Reader | no
VM | 0%
```

Extensions Info:
```
Extension | Author (truncated) | Version
-- | -- | --
remote-containers | ms- | 0.134.1
remote-ssh | ms- | 0.51.0
remote-ssh-edit | ms- | 0.51.0
remote-wsl | ms- | 0.44.5
vscode-remote-extensionpack | ms- | 0.20.0
vscode-eslint | dba | 2.1.8
xml | Dot | 2.5.1
prettier-vscode | esb | 5.5.0
terraform | has | 2.2.2
beautify | Hoo | 1.5.0
character-position | kyr | 1.0.0
pyright | ms- | 1.1.65
python | ms- | 2020.8.105369
```

**Comment by erictraut** [comment]
Yes, this is correct behavior. The type parameter for a `List` (or any mutable container class) is invariant, so the class type must match exactly. For more information about this, refer to [this documentation](https://github.com/microsoft/pyright/blob/master/docs/type-concepts.md#generic-types).
